lib/cli/src/commands/create_exe.rs

```diff
@@ -344,1043 +256,1624 @@ impl CreateExe {
+        #[cfg(unix)]
+        let system_path_var = std::process::Command::new("getconf")
+            .args(&["PATH"])
+            .output()
+            .map(|output| output.stdout)
+            .unwrap_or_default();
+        let retval = if let Some(p) = path {
+            if p.exists() {
+                p
+            } else {
+                return Err(anyhow!("Could not find `zig` binary in {}.", p.display()));
+            }
         } else {
-            command
+            let mut retval = None;
+            for mut p in split_paths(&path_var).chain(split_paths(
+                #[cfg(unix)]
+                {
+                    &OsStr::from_bytes(&system_path_var[..])
+                },
+                #[cfg(not(unix))]
+                {
+                    OsStr::new("")
+                },
+            )) {
+                p.push(get_zig_exe_str());
+                if p.exists() {
+                    retval = Some(p);
+                    break;
+                }
+            }
+            retval.ok_or_else(|| anyhow!("Could not find `zig` binary in PATH."))?
         };
-        // Add libraries required per platform.
-        // We need userenv, sockets (Ws2_32), advapi32 for some system calls and bcrypt for random numbers.
-        #[cfg(windows)]
-        let command = command
-            .arg("-luserenv")
-            .arg("-lWs2_32")
-            .arg("-ladvapi32")
-            .arg("-lbcrypt");
-        // On unix we need dlopen-related symbols, libmath for a few things, and pthreads.
-        #[cfg(not(windows))]
-        let command = command.arg("-ldl").arg("-lm").arg("-pthread");
-        let link_against_extra_libs = self
-            .additional_libraries
+
+        let version = std::process::Command::new(&retval)
+            .arg("version")
+            .output()
+            .with_context(|| {
+                format!(
+                    "Could not execute `zig` binary at path `{}`",
+                    retval.display()
+                )
+            })?
+            .stdout;
+        let version_slice = if let Some(pos) = version
             .iter()
-            .map(|lib| format!("-l{}", lib));
-        let command = command.args(link_against_extra_libs);
-        let command = command.arg("-o").arg(&self.output_path);
-        let output = command.output()?;
-
-        if !output.status.success() {
-            bail!(
-                "linking failed with: stdout: {}\n\nstderr: {}",
-                std::str::from_utf8(&output.stdout)
-                    .expect("stdout is not utf8! need to handle arbitrary bytes"),
-                std::str::from_utf8(&output.stderr)
-                    .expect("stderr is not utf8! need to handle arbitrary bytes")
-            );
+            .position(|c| !(c.is_ascii_digit() || (*c == b'.')))
+        {
+            &version[..pos]
+        } else {
+            &version[..]
+        };
+
+        if version_slice < b"0.10.0".as_ref() {
```

Comment:
Should we use the `semver` library for comparing version numbers?

Reply:
Done in https://github.com/wasmerio/wasmer/pull/3430/commits/c8e7185917473153836709e1ecd54ad97a1dd812